File: Week 5/5. Parameter/ros2_ws/src/pkg_py/pkg_py/number_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from example_interfaces.msg import Int64

class NumberPublisher(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.declare_parameter("num",0)
        self.num = self.get_parameter("num").value

        self.publisher = self.create_publisher(Int64, "/number", 10)
        self.timer = self.create_timer(1.0, self.publish_number)

        self.get_logger().info(f"Number has started counting from {self.num}")

    def publish_number(self):
        msg = Int64()
        self.num += 1
        msg.data = self.num
        self.publisher.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: Week 5/5. Parameter/ros2_ws/src/pkg_cpp/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher() : Node("number_publisher")
    {
        this->declare_parameter("number", 1);
        number = this->get_parameter("number").as_int();

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("/number", 10);
        timer_ = this->create_wall_timer(1s, std::bind(&NumberPublisher::publish_counter, this));

        RCLCPP_INFO(this->get_logger(), "The number counter has started from %d.",number);
    }

private:
    void publish_counter()
    {
        msg.data = number;
        number++;
        publisher_->publish(msg);
    }

    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    example_interfaces::msg::Int64 msg;
    int number;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<NumberPublisher>());
    rclcpp::shutdown();
    return 0;
}